### PR TITLE
base model attributes interaction

### PR DESCRIPTION
**Pull request by kazalt**

### CODE REVIEW

File: test/base.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { expect } from 'chai';
+import Base from '../base.js';
+
+describe('Base', () => {
```
Comment (oudirichi) [inline test/base.js]: the set method is not tested and I think the set did not really change the value in attributes (never affected)

File: attribute.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  setValue(newValue, oldValue) {
+    if (!isEqual(oldValue, newValue)) {
+      this.hasChanged = !isEqual(this.getOriginalValue(), newValue);
+      this.isDirty = true;
+      this._parent.isDirty = true;
+    }
+
+    return this.parse(newValue);
```
Comment (oudirichi) [inline attribute.js]: missing the assignation into the right attribute

File: base.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import Attribute from './attribute';
+
+class BaseModel {
```
Comment (oudirichi) [inline base.js]: what would be nice is:
the proxy are set directly on the model so we can do myInstance.name = "something", this way the standard of model are respected. furthermore this value is set on attributes (plain object). This way we could print all values of attributes. At this moment we need to use the get to know the value when debugging. myInstance.attributes.name.get() (impossible to know by outputting the name directly).

Comment (oudirichi) [reply base.js]: set by a proxy, this way we still can trigger validations and callbacks

File: attribute.js (hunk 2 above)
Comment (kazalt) [reply attribute.js]: setValue is just an exposed method used by the constructor for the `set` method (line 49)

File: test/base.js (hunk 1 above)
Comment (kazalt) [reply test/base.js]: the set method ? in `base.js` ? there is no such method

**Comment by oudirichi** [comment]
I approve the last PR

**Pull request merged**